torch/_dynamo/variables/functions.py

```diff
@@ -444,6 +444,12 @@ def _setup_exception(self, tx, exc):
             # exception is raised again.
             tracer.exception_handler(e)
 
+    def _is_inside_try_finally(self, tracer):
```

Comment:
To confirm, what you're checking is is if the tracer is currently inside the try: block of a try-finally?

Reply:
> To confirm, what you're checking is is if the tracer is currently inside the try: block of a try-finally?

Almost. I'm checking if there is a finally block.


I found a test case that breaks this. 

```python
def test_throw_aaa(self):
    z = 0
    def whoo(t):
        nonlocal z
        try:
            z += 1
            yield t.sin()
        except ValueError:
            z += 1
            yield t.cos()
        z += 1

    @torch.compile(backend="eager", fullgraph=True)
    def fn(t):
        gen = whoo(t)
        a = next(gen)
        b = gen.throw(ValueError)
        return a + b

    t = torch.randn(2)
    y = fn(t)
    self.assertEqual(y, t.sin() + t.cos())
    self.assertEqual(z, 2)
```

Reply:
Fixed!